### PR TITLE
Correctly associate source with a topic

### PR DESCRIPTION
**Pull request by llawall**
Minor refactor of building a topology. Ensures sources are kept
in a dictionary so a stream task can call process on correct source
for the current record.

### CODE REVIEW

File: winton_kafka_streams/processor/topology.py [hunk 1]
```diff
@@ -31,46 +31,63 @@ def __repr__(self):
...
-        self.processors = _processors
-        self.sinks = _sinks
-        self.state_stores = _state_stores
+    """
+    A realised instance of a toplogy
+
+    """
+    def __init__(self, sources, processors, sinks, state_stores):
+        self.nodes = {}
+        self.sources = {}
```
Comment (llawall) [inline winton_kafka_streams/processor/topology.py]: Sources now kept in a dictionary to ensure we can associate a topic with the correct processor in stream task.

File: winton_kafka_streams/processor/_stream_task.py [hunk 2]
```diff
@@ -85,9 +85,8 @@ def process(self):
         self.context.currentRecord = record
         self.current_timestamp = self.timestamp_extractor.extract(record, self.current_timestamp)
 
-        # TODO: FIXME-  assumes only one topic (next two lines)
-        self.context.currentNode = self.topology.sources[0]
-        self.topology.sources[0].process(record.key(), record.value())
+        self.context.currentNode = self.topology.sources[record.topic()]
+        self.topology.sources[record.topic()].process(record.key(), record.value())
```
Comment (llawall) [inline winton_kafka_streams/processor/_stream_task.py]: Process a record with the correct source.

**Comment by llawall** [comment]
This commit is pulled out of some changes I was making to implement a proof of concept DSL for the project. It should (I hope) be useful to @phoenix10k

File: examples/binning/binning.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import pandas as pd
 from winton_kafka_streams.processor import BaseProcessor, TopologyBuilder
 import winton_kafka_streams.kafka_config as kafka_config
-import winton_kafka_streams.kafka_stream as kafka_stream
+import winton_kafka_streams.kafka_streams as kafka_streams
```
Comment (satlank) [inline examples/binning/binning.py]: Oh, this is broken at the moment? (or is the diff view confusing me again?)

**Comment by phoenix10k** [comment]
Great, I'm hoping to submit a PR adding a test/example based on https://github.com/confluentinc/examples/blob/3.2.x/kafka-streams/src/test/java/io/confluent/examples/streams/StreamToStreamJoinIntegrationTest.java soon (and any code required to get it to work :) )

File: examples/binning/binning.py (hunk 3 above)
Comment (llawall) [inline examples/binning/binning.py]: I'm not sure what you mean exactly. This link on master shows that the package name, module name and class name are all plural now:

https://github.com/wintoncode/winton-kafka-streams/blob/master/winton_kafka_streams/kafka_streams.py#L14

**Comment by llawall** [comment]
@phoenix10k Excellent - looking forward to it. Are you intending to put together the DSL pieces too or will you implement the example using the more fundamental processsors?

**Comment by llawall** [comment]
@satlank Fair point about calling the private function. I'm not 100% happy with the way this is now either though I prefer it slightly more than before where nodes had to be passed into every one of the builder functions.

File: examples/binning/binning.py (hunk 3 above)
Comment (satlank) [inline examples/binning/binning.py]: Yes. But if the rename doesn't happen in this PR (and afaict, it doesn't), then is master working?

Comment (satlank) [inline examples/binning/binning.py]: ... or the diff isn't showing me what I think it is showing me.

**Comment by phoenix10k** [comment]
I should have said soon™. How is the DSL coming along? I think it would be nice to use it for this simple example if we know pretty much what we want it to look like already, but I'm happy to stick to the processor API too if we're still a little way off, and I don't want to bite off too much at once.

File: examples/binning/binning.py (hunk 3 above)
Comment (ah-) [inline examples/binning/binning.py]: Should be fixed in master now. Forgot to make examples import plural version.

**Pull request merged**